### ISSUE TITLE
Function for getting Unit with friends around

### ISSUE THREAD

**Issue by GabbzGit** [MrTheSoulz/NerdPack#63]
This request is coming from my need for Holy Prism but are generic.
I need to get a enemy unit that meets some criteria.

Design:
Parse through all enemies and check if there are x number of friendly units that are beneath a health level. Note that it should give me the best and not just the first that meets the requirements. For example Holy Prism is targeting 5 allies within 15 yards but it is ok to cast if there is 3.

So i want the possibility to set requirement as 3 beneath 80% but want the best unit with the most injured allies around.

I also want to have similiar for getting the best ally to cast on to heal around and i also want the best ally to cast to damage enemies around.

Basically
* Ally with x number of enemies around
* Ally with x number of allies around
* Enemy with x number of enemies around
* Enemy with x number of allies around.

**Comment by MrTheSoulz** [comment]
I suggest using NeP.fakeUnits.Add(NAME, FUNC) to create new units with those needs.
